### DISCUSSION TITLE
Mania skin editor should let user define lanes, and then assemble keymodes out of them.

### DISCUSSION THREAD

**Discussion by AkiraMorilas** [ppy/osu#29961] [Ideas]
From the lazer update videos I'm assuming that right now skinning and things regarding the GUI skin editor are being worked on, so I think it's an opportunity to do things differently. `skin.ini` has many gotchas(like `Colour#:` being 1-indexed when everything else related is 0-indexed) and shortcomings(like lack of per-key judgement line making skinning distinct receptors a lengthy process involving trial and error with `KeyImage#:` and how it scales). I wrote a script for a whole video about them, but after some time I managed to compress all but two points of that video into the sentence in the title.

Here's a quick mock-up:
![lane](https://github.com/user-attachments/assets/270319eb-1539-41fd-8ff2-8339a77e6362)
![keymode](https://github.com/user-attachments/assets/f13a1651-4811-43ac-9731-b580ee4a3dac)

The benefit of this over copying the way skin.ini currently does things(separately specifying every single image for every single keymode)is simplification and reducing work from the users point of view. I would know, because back when I was actively developing my skin, I wrote a python script to do it for me from a preset .txt file.

**Comment by Tenexxt** [comment]
Lane editor in the game would be so awsome not gonna lie.
I don't even use other skins than argon, but if i would ever get to skinning i can see this being very helpfull.
I think that more and more things should be accesible to edit in the game and not just skin.ini, would make the skin editor way more usefull.

**Comment by AkiraMorilas** [comment]
Now that the newest osu!update video has peppy saying that skinning is the new focus, I think this proposal is worth revisiting.

**Reply by peppy** [reply]
mania is its own isolated issue with skinning, and would come as a follow-up effort if anything. don't get too hopeful for immediate action.
